independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/JarResource.java

```diff
@@ -30,25 +26,22 @@ public class JarResource implements ClassLoadingResource {
     private final ManifestInfo manifestInfo;
     private final Path jarPath;
 
-    private final Lock readLock;
-    private final Lock writeLock;
-
     private volatile ProtectionDomain protectionDomain;
 
-    //Guarded by the read/write lock; open/close operations on the JarFile require the exclusive lock,
-    //while using an existing open reference can use the shared lock.
-    //If a lock is acquired, and as long as it's owned, we ensure that the zipFile reference
-    //points to an open JarFile instance, and read operations are valid.
-    //To close the jar, the exclusive lock must be owned, and reference will be set to null before releasing it.
-    //Likewise, opening a JarFile requires the exclusive lock.
-    private volatile JarFile zipFile;
+    private final AtomicReference<JarFileReference> jarFileReference = new AtomicReference<>(null);
```

Comment:
@franz1981 not that it makes any difference here, but it isn't a tiny bit faster  to do `new AtomicReference<>()`?

Reply:
Fixed, thanks for suggestion.